### DISCUSSION TITLE
GA402RK - Power limits per mode

### DISCUSSION THREAD

**Discussion by lucidnx** [seerge/g-helper#3377] [General]
Hello! I would like to know what are stock limits in each of default modes (Eco, Balanced and Turbo) on G14 2022
1. max temperature of CPU/GPU while laptop is open
2. max temp while it's closed (laptop in stand, using external monitor)
3. stock power limits Platform/CPU

I am asking because if I create exactly same mode (power limits, fan curves, etc..) as Turbo, It's running cooler and I would say more smooth in longer gameplay

**Comment by seerge** [comment]
@lucidnx hello, the only way to get them is experimentally. I.e. by running Cinebench and monitoring power draw.

You can also try to run ryzenadj app with -i param, for some AMD CPUs it can get SPL, SPPT, FPPT and CPU throttle temp.

For all GPUs (even Nvidia) Asus seem to set 87 as throttle temp by default

**Reply by lucidnx** [reply]
Thanks! I wonder then, why my laptop always go up to 96 celsius on CPU, according to hwinfo, there's no thermal throttling...
Since day 0, I think my laptop is broken.. weird behavior all the time, unexpected things happening, random crashing but then, well it can be really efficient (screen)..
![image](https://github.com/user-attachments/assets/2d842eca-f94b-48f8-8e2d-d99dbde2404e)

**Reply by seerge** [reply]
@lucidnx usually 96c is a throttle temp for CPUs for Balanced and Turbo, for Silent it could be lower. So ryzenadj can help with checking

**Reply by lucidnx** [reply]
@seerge downloading, I wonder what it will throw on me :))

**Reply by lucidnx** [reply]
well..  results are somewhat as expected, even turning off windows security stuff (memory integrity, virtualization based protections, etc..) did not helped.
![image](https://github.com/user-attachments/assets/1e4a75e0-a6b2-4b37-9dbf-57621ceeabc3)


will try on linux, maybe it will give me some results..

**Reply by lucidnx** [reply]
@seerge if you can help, I don't see anything that will affect temps the way I mentioned.. please, do you see something?
// edit: pasted wrong output, now I see, performance temp limit 96, balanced 93, eco 90. I guess that custom profiles created in ghelper are using balanced as default?

on linux, profiles changed using asusctl, system mostly idle:

Balanced PD

CPU Family: Rembrandt
SMU BIOS Interface Version: 18
Version: v0.16.0 
PM Table Version: 450005
|        Name         |   Value   |     Parameter      |
|---------------------|-----------|--------------------|
| STAPM LIMIT         |    79.964 | stapm-limit        |
| STAPM VALUE         |     4.922 |                    |
| PPT LIMIT FAST      |    80.000 | fast-limit         |
| PPT VALUE FAST      |     8.572 |                    |
| PPT LIMIT SLOW      |    80.000 | slow-limit         |
| PPT VALUE SLOW      |     5.140 |                    |
| StapmTimeConst      |       nan | stapm-time         |
| SlowPPTTimeConst    |       nan | slow-time          |
| PPT LIMIT APU       |    35.000 | apu-slow-limit     |
| PPT VALUE APU       |    23.072 |                    |
| TDC LIMIT VDD       |    65.000 | vrm-current        |
| TDC VALUE VDD       |     2.844 |                    |
| TDC LIMIT SOC       |    15.000 | vrmsoc-current     |
| TDC VALUE SOC       |     1.576 |                    |
| EDC LIMIT VDD       |   140.000 | vrmmax-current     |
| EDC VALUE VDD       |    72.493 |                    |
| EDC LIMIT SOC       |    20.000 | vrmsocmax-current  |
| EDC VALUE SOC       |     0.069 |                    |
| THM LIMIT CORE      |    93.000 | tctl-temp          |
| THM VALUE CORE      |    47.767 |                    |
| STT LIMIT APU       |    36.000 | apu-skin-temp      |
| STT VALUE APU       |    32.510 |                    |
| STT LIMIT dGPU      |    46.000 | dgpu-skin-temp     |
| STT VALUE dGPU      |    36.190 |                    |
| CCLK Boost SETPOINT |       nan | power-saving /     |
| CCLK BUSY VALUE     |       nan | max-performance    |

Performance PD

CPU Family: Rembrandt
SMU BIOS Interface Version: 18
Version: v0.16.0 
PM Table Version: 450005
|        Name         |   Value   |     Parameter      |
|---------------------|-----------|--------------------|
| STAPM LIMIT         |    80.000 | stapm-limit        |
| STAPM VALUE         |     5.332 |                    |
| PPT LIMIT FAST      |    80.000 | fast-limit         |
| PPT VALUE FAST      |    15.557 |                    |
| PPT LIMIT SLOW      |    80.000 | slow-limit         |
| PPT VALUE SLOW      |     5.413 |                    |
| StapmTimeConst      |       nan | stapm-time         |
| SlowPPTTimeConst    |       nan | slow-time          |
| PPT LIMIT APU       |    35.000 | apu-slow-limit     |
| PPT VALUE APU       |    23.072 |                    |
| TDC LIMIT VDD       |    65.000 | vrm-current        |
| TDC VALUE VDD       |     3.472 |                    |
| TDC LIMIT SOC       |    15.000 | vrmsoc-current     |
| TDC VALUE SOC       |     1.504 |                    |
| EDC LIMIT VDD       |   140.000 | vrmmax-current     |
| EDC VALUE VDD       |   120.457 |                    |
| EDC LIMIT SOC       |    20.000 | vrmsocmax-current  |
| EDC VALUE SOC       |     0.071 |                    |
| THM LIMIT CORE      |    96.000 | tctl-temp          |
| THM VALUE CORE      |    54.821 |                    |
| STT LIMIT APU       |    36.000 | apu-skin-temp      |
| STT VALUE APU       |    31.120 |                    |
| STT LIMIT dGPU      |    47.000 | dgpu-skin-temp     |
| STT VALUE dGPU      |    33.561 |                    |
| CCLK Boost SETPOINT |       nan | power-saving /     |
| CCLK BUSY VALUE     |       nan | max-performance    |

Quiet PD

CPU Family: Rembrandt
SMU BIOS Interface Version: 18
Version: v0.16.0 
PM Table Version: 450005
|        Name         |   Value   |     Parameter      |
|---------------------|-----------|--------------------|
| STAPM LIMIT         |    77.133 | stapm-limit        |
| STAPM VALUE         |     6.504 |                    |
| PPT LIMIT FAST      |    80.000 | fast-limit         |
| PPT VALUE FAST      |     7.923 |                    |
| PPT LIMIT SLOW      |    80.000 | slow-limit         |
| PPT VALUE SLOW      |     5.976 |                    |
| StapmTimeConst      |       nan | stapm-time         |
| SlowPPTTimeConst    |       nan | slow-time          |
| PPT LIMIT APU       |    35.000 | apu-slow-limit     |
| PPT VALUE APU       |    23.072 |                    |
| TDC LIMIT VDD       |    65.000 | vrm-current        |
| TDC VALUE VDD       |     2.912 |                    |
| TDC LIMIT SOC       |    15.000 | vrmsoc-current     |
| TDC VALUE SOC       |     1.466 |                    |
| EDC LIMIT VDD       |   140.000 | vrmmax-current     |
| EDC VALUE VDD       |   102.741 |                    |
| EDC LIMIT SOC       |    20.000 | vrmsocmax-current  |
| EDC VALUE SOC       |     0.067 |                    |
| THM LIMIT CORE      |    90.000 | tctl-temp          |
| THM VALUE CORE      |    47.427 |                    |
| STT LIMIT APU       |    36.000 | apu-skin-temp      |
| STT VALUE APU       |    32.792 |                    |
| STT LIMIT dGPU      |    46.000 | dgpu-skin-temp     |
| STT VALUE dGPU      |    36.489 |                    |
| CCLK Boost SETPOINT |       nan | power-saving /     |
| CCLK BUSY VALUE     |       nan | max-performance    |

Balanced Batt

CPU Family: Rembrandt
SMU BIOS Interface Version: 18
Version: v0.16.0 
PM Table Version: 450005
|        Name         |   Value   |     Parameter      |
|---------------------|-----------|--------------------|
| STAPM LIMIT         |    71.610 | stapm-limit        |
| STAPM VALUE         |     4.632 |                    |
| PPT LIMIT FAST      |    80.000 | fast-limit         |
| PPT VALUE FAST      |     6.497 |                    |
| PPT LIMIT SLOW      |    80.000 | slow-limit         |
| PPT VALUE SLOW      |     4.544 |                    |
| StapmTimeConst      |       nan | stapm-time         |
| SlowPPTTimeConst    |       nan | slow-time          |
| PPT LIMIT APU       |    45.000 | apu-slow-limit     |
| PPT VALUE APU       |    23.072 |                    |
| TDC LIMIT VDD       |    65.000 | vrm-current        |
| TDC VALUE VDD       |     3.010 |                    |
| TDC LIMIT SOC       |    15.000 | vrmsoc-current     |
| TDC VALUE SOC       |     1.412 |                    |
| EDC LIMIT VDD       |   140.000 | vrmmax-current     |
| EDC VALUE VDD       |    97.155 |                    |
| EDC LIMIT SOC       |    20.000 | vrmsocmax-current  |
| EDC VALUE SOC       |     0.074 |                    |
| THM LIMIT CORE      |    90.000 | tctl-temp          |
| THM VALUE CORE      |    48.921 |                    |
| STT LIMIT APU       |    36.000 | apu-skin-temp      |
| STT VALUE APU       |    34.446 |                    |
| STT LIMIT dGPU      |    46.000 | dgpu-skin-temp     |
| STT VALUE dGPU      |    40.707 |                    |
| CCLK Boost SETPOINT |       nan | power-saving /     |
| CCLK BUSY VALUE     |       nan | max-performance    |

Performance Batt

CPU Family: Rembrandt
SMU BIOS Interface Version: 18
Version: v0.16.0 
PM Table Version: 450005
|        Name         |   Value   |     Parameter      |
|---------------------|-----------|--------------------|
| STAPM LIMIT         |    86.503 | stapm-limit        |
| STAPM VALUE         |     6.428 |                    |
| PPT LIMIT FAST      |    80.000 | fast-limit         |
| PPT VALUE FAST      |    13.822 |                    |
| PPT LIMIT SLOW      |    80.000 | slow-limit         |
| PPT VALUE SLOW      |     6.541 |                    |
| StapmTimeConst      |       nan | stapm-time         |
| SlowPPTTimeConst    |       nan | slow-time          |
| PPT LIMIT APU       |    80.000 | apu-slow-limit     |
| PPT VALUE APU       |    23.072 |                    |
| TDC LIMIT VDD       |    65.000 | vrm-current        |
| TDC VALUE VDD       |     4.349 |                    |
| TDC LIMIT SOC       |    15.000 | vrmsoc-current     |
| TDC VALUE SOC       |     1.449 |                    |
| EDC LIMIT VDD       |   140.000 | vrmmax-current     |
| EDC VALUE VDD       |   124.754 |                    |
| EDC LIMIT SOC       |    20.000 | vrmsocmax-current  |
| EDC VALUE SOC       |     0.074 |                    |
| THM LIMIT CORE      |    96.000 | tctl-temp          |
| THM VALUE CORE      |    53.200 |                    |
| STT LIMIT APU       |    36.000 | apu-skin-temp      |
| STT VALUE APU       |    34.014 |                    |
| STT LIMIT dGPU      |    47.000 | dgpu-skin-temp     |
| STT VALUE dGPU      |    39.422 |                    |
| CCLK Boost SETPOINT |       nan | power-saving /     |
| CCLK BUSY VALUE     |       nan | max-performance    |

Quiet Batt

CPU Family: Rembrandt
SMU BIOS Interface Version: 18
Version: v0.16.0 
PM Table Version: 450005
|        Name         |   Value   |     Parameter      |
|---------------------|-----------|--------------------|
| STAPM LIMIT         |    78.324 | stapm-limit        |
| STAPM VALUE         |     4.756 |                    |
| PPT LIMIT FAST      |    80.000 | fast-limit         |
| PPT VALUE FAST      |     7.497 |                    |
| PPT LIMIT SLOW      |    80.000 | slow-limit         |
| PPT VALUE SLOW      |     5.763 |                    |
| StapmTimeConst      |       nan | stapm-time         |
| SlowPPTTimeConst    |       nan | slow-time          |
| PPT LIMIT APU       |    45.000 | apu-slow-limit     |
| PPT VALUE APU       |    23.072 |                    |
| TDC LIMIT VDD       |    65.000 | vrm-current        |
| TDC VALUE VDD       |     2.972 |                    |
| TDC LIMIT SOC       |    15.000 | vrmsoc-current     |
| TDC VALUE SOC       |     1.358 |                    |
| EDC LIMIT VDD       |   140.000 | vrmmax-current     |
| EDC VALUE VDD       |    82.848 |                    |
| EDC LIMIT SOC       |    20.000 | vrmsocmax-current  |
| EDC VALUE SOC       |     0.068 |                    |
| THM LIMIT CORE      |    90.000 | tctl-temp          |
| THM VALUE CORE      |    47.448 |                    |
| STT LIMIT APU       |    36.000 | apu-skin-temp      |
| STT VALUE APU       |    33.318 |                    |
| STT LIMIT dGPU      |    46.000 | dgpu-skin-temp     |
| STT VALUE dGPU      |    37.483 |                    |
| CCLK Boost SETPOINT |       nan | power-saving /     |
| CCLK BUSY VALUE     |       nan | max-performance    |
